pkg/kubectl/stop.go

```diff
@@ -167,6 +176,53 @@ func (reaper *ReplicationControllerReaper) Stop(namespace, name string, timeout
 	return fmt.Sprintf("%s stopped", name), nil
 }
 
+func (reaper *DaemonSetReaper) Stop(namespace, name string, timeout time.Duration, gracePeriod *api.DeleteOptions) (string, error) {
+	daemon, err := reaper.Experimental().DaemonSets(namespace).Get(name)
+	if err != nil {
+		return "", err
+	}
+
+	// Update the daemon set to select for a non-existent NodeName.
+	// The daemon set controller will then kill all the daemon pods corresponding to daemon set.
+	nodes, err := reaper.Nodes().List(labels.Everything(), fields.Everything())
+	if err != nil {
+		return "", err
+	}
+	var fuzzer = fuzz.New()
+	var nameExists bool
+
+	var nodeName string
+	fuzzer.Fuzz(&nodeName)
+	nameExists = false
+	for _, node := range nodes.Items {
+		nameExists = nameExists || node.Name == nodeName
+	}
+	if nameExists {
+		// Probability of reaching here is extremely low, most likely indicates a programming bug/library error.
+		return "", fmt.Errorf("Name collision. Failed to stop node.")
+	}
+
+	daemon.Spec.Template.Spec.NodeName = nodeName
+
+	reaper.Experimental().DaemonSets(namespace).Update(daemon)
```

Comment:
Unfortunately and empty node name actually means schedule on all nodes.

https://github.com/kubernetes/kubernetes/blob/master/pkg/api/types.go#L979-L981